libpages/server.go

```diff
@@ -287,6 +287,27 @@ func (s *Server) logRequest(sri *ServedRequestInfo, requestPath string) {
 	)
 }
 
+func (s *Server) setCommonResponseHeaders(w http.ResponseWriter) {
+	// Since http.FileServer already sets MIME type properly, disable MIME type
+	// sniffing on browser-side. This would prevent e.g. an attack where a
+	// malicious html file with .jpg suffix being executed by browser without
+	// site visitors' awareness. References:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options
+	// https://helmetjs.github.io/docs/dont-sniff-mimetype/
+	w.Header().Set("X-Content-Type-Options", "nosniff")
```

Comment:
I'm actually leaning towards not having this header, for three reasons:

1) Keybase Pages doesn't have visitor-uploaded content; all content is provided by the user who hosts with us. This means an evil user Evan has to already be hosting with us in order to launch an attack using this, on his own site. For example, Evan hosts an html under jpg, and sends a link to Alice, in which case we are hosting hostile content from Evan to Alice, when Evan controls the domain. This is pretty rare, and arguably as a hosting provider, we shouldn't be worried about this kind of attack.

2) This attack in example didn't work in my Chrome without the added `nosniff` header, so modern browsers might have already been doing the right thing.

3) As mentioned on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options) A naive implementation of `nosniff` may break image rendering, and it was fixed in standard [only recently](https://github.com/whatwg/fetch/issues/395), so enabling this header globally for all returned content may still break some use cases.

That being said, keybase.pub has this header returned for both webpage and images. So removing this would be a change.

cc @malgorithms

Reply:
cool, your call on this since I'm not sure.